lib/peep/storage/atomics.ex

```diff
@@ -0,0 +1,69 @@
+defmodule Peep.Storage.Atomics do
+  @moduledoc false
+
+  defstruct [
+    :num_buckets,
+    :buckets,
+    :sum,
+    :above_max,
+    :bucket_calculator
+  ]
+
+  def new(%Telemetry.Metrics.Distribution{} = metric) do
+    {bucket_calculator, config} = Peep.Buckets.config(metric)
+    num_buckets = bucket_calculator.number_of_buckets(config)
+    buckets = :atomics.new(num_buckets, signed: false)
+    sum = :atomics.new(1, signed: false)
+    above_max = :atomics.new(1, signed: false)
```

Comment:
I'm assuming you don't allow for negative values to be placed in the histogram? Otherwise, you'd also need a `below_min` here. That being said, that use case should be pretty rare for histograms.

Reply:
Good point!

The Prometheus docs discourage recording negative values in distributions because they mess up the `_sum` metric, which is often used for tracking rates of change. The docs recommend storing positive and negative values in separate distributions/timeseries, but they don't explicitly forbid it.

The quick and dirty approach I chose for Peep.Buckets.Exponential is to increment the smallest valued bucket for all values less than 1.

```elixir
   @impl true
-  def bucket_for(value, _) when value == 0 do
+  def bucket_for(value, _) when value < 1 do
    0
   end
```